require_ansible_source | missing libffi

```
      Installing collected packages: idna, pyasn1, setuptools, enum34, ipaddress, pycparser, cffi, cryptography, paramiko, MarkupSafe, Jinja2
         Found existing installation: setuptools 3.3
           Uninstalling setuptools-3.3:
             Successfully uninstalled setuptools-3.3
         Running setup.py install for pycparser ... done
         Running setup.py install for cffi ... error
           Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-0I0TyS/cffi/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-AYJUzB-record/install-record.txt --single-version-externally-managed --compile:
           running install
           running build
           running build_py
           creating build
           creating build/lib.linux-x86_64-2.7
           creating build/lib.linux-x86_64-2.7/cffi
           copying cffi/backend_ctypes.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/__init__.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/vengine_cpy.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/api.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/recompiler.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/verifier.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/model.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/lock.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/cparser.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/cffi_opcode.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/setuptools_ext.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/ffiplatform.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/vengine_gen.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/commontypes.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/gc_weakref.py -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/_cffi_include.h -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/parse_c_type.h -> build/lib.linux-x86_64-2.7/cffi
           copying cffi/_embedding.h -> build/lib.linux-x86_64-2.7/cffi
           running build_ext
           building '_cffi_backend' extension
           creating build/temp.linux-x86_64-2.7
           creating build/temp.linux-x86_64-2.7/c
           x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DUSE__THREAD -I/usr/include/ffi -I/usr/include/libffi -I/usr/include/python2.7 -c c/_cffi_backend.c -o build/temp.linux-x86_64-2.7/c/_cffi_backend.o
           c/_cffi_backend.c:15:17: fatal error: ffi.h: No such file or directory
            #include <ffi.h>
                     ^
           compilation terminated.
           error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
           
           ----------------------------------------
       Command "/usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-0I0TyS/cffi/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-AYJUzB-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-0I0TyS/cffi/
       /usr/local/lib/python2.7/dist-packages/pip-8.1.2-py2.7.egg/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
         InsecurePlatformWarning
       Transferring files to <default-ubuntu-trusty64>
       Going to invoke ansible-playbook with: ANSIBLE_ROLES_PATH=/tmp/kitchen/roles sudo -E -s  PATH=/tmp/kitchen/ansible/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games PYTHONPATH=/tmp/kitchen/ansible/lib MANPATH=/tmp/kitchen/ansible/docs/man /tmp/kitchen/ansible/bin/ansible-playbook -i /tmp/kitchen/hosts  -c local -M /tmp/kitchen/modules -vv        /tmp/kitchen/default.yml
       Traceback (most recent call last):
         File "/tmp/kitchen/ansible/bin/ansible-playbook", line 44, in <module>
           import ansible.playbook
         File "/tmp/kitchen/ansible/lib/ansible/playbook/__init__.py", line 20, in <module>
           import ansible.runner
         File "/tmp/kitchen/ansible/lib/ansible/runner/__init__.py", line 32, in <module>
           import jinja2
       ImportError: No module named jinja2
```

my `.kitchen.yml`

```
---
driver:
  name: <%= ENV['KITCHEN_DRIVER'] || 'vagrant' %>
  provider: <%= ENV['KITCHEN_PROVIDER'] || 'virtualbox' %>
  require_chef_omnibus: false
  customize:
    memory: 256

provisioner:
  hosts: test-kitchen
  name: ansible_playbook
  require_ansible_source: true
  ansible_source_rev: <%= ENV['ANSIBLE_VERSION'] || 'v1.9.2-1' %>
  require_chef_for_busser: false
  require_ruby_for_busser: false
  ansible_verbosity: 2
  ansible_verbose: true
  idempotency_test: true
  update_package_repos: false

verifier:
  name: inspec

platforms:
  - name: ubuntu/trusty64

suites:
  - name: default
```

--- a/lib/kitchen/provisioner/ansible_playbook.rb
+++ b/lib/kitchen/provisioner/ansible_playbook.rb
@@ -334,14 +334,13 @@ module Kitchen
           return config[:ansible_playbook_command]
         else
 
-          cmd = ansible_command('ansible-playbook')
-          if config[:require_ansible_source]
+          if config[:require_ansible_source] && !config[:ansible_binary_path]
             # this is an ugly hack to get around the fact that extra vars uses ' and "
-            cmd = ansible_command("PATH=#{config[:root_path]}/ansible/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games PYTHONPATH=#{config[:root_path]}/ansible/lib MANPATH=#{config[:root_path]}/ansible/docs/man #{config[:root_path]}/ansible/bin/ansible-playbook")
-          end
-
-          if config[:ansible_binary_path]
+            cmd = ansible_command("PATH=#{config[:root_path]}/ansible/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games PYTHONPATH=#{config[:root_path]}/ansible/lib MANPATH=#{config[:root_path]}/ansible/docs/man ansible-playbook")
+          elsif config[:ansible_binary_path]
             cmd = ansible_command("#{config[:ansible_binary_path]}/ansible-playbook")
+          else
+            cmd = ansible_command('ansible-playbook')
           end
 
           cmd = "HTTPS_PROXY=#{https_proxy} #{cmd}" if https_proxy
@@ -423,15 +422,15 @@ module Kitchen
           if [ -f /etc/centos-release ] || [ -f /etc/redhat-release ]; then
             #{Kitchen::Provisioner::Ansible::Os::Redhat.new('redhat', config).install_epel_repo}
             #{update_packages_redhat_cmd}
-            #{sudo_env('yum')} -y install libselinux-python python2-devel git python-setuptools python-setuptools-dev
+            #{sudo_env('yum')} -y install libselinux-python python2-devel git python-setuptools python-setuptools-dev libffi-devel libssl-devel
           else
             if [ -f /etc/SUSE-brand ] || [ -f /etc/SuSE-release ]; then
               #{sudo_env('zypper')} ar #{python_sles_repo}
               #{update_packages_suse_cmd}
-              #{sudo_env('zypper')} --non-interactive install python python-devel git python-setuptools python-pip python-six libyaml-devel
+              #{sudo_env('zypper')} --non-interactive install python python-devel git python-setuptools python-pip python-six libyaml-devel libffi-devel libopenssl-devel
             else
               #{update_packages_debian_cmd}
-              #{sudo_env('apt-get')} -y install git python python-setuptools build-essential python-dev
+              #{sudo_env('apt-get')} -y install git python python-setuptools build-essential python-dev libffi-dev libssl-dev
             fi
           fi
 

